include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -0,0 +1,132 @@
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_parallel_backend_sycl_radix_sort_one_wg_H
+#define _ONEDPL_parallel_backend_sycl_radix_sort_one_wg_H
+
+//#include <limits>
+
+//#include "sycl_defs.h"
+//#include "parallel_backend_sycl_utils.h"
+//#include "execution_sycl_defs.h"
+
+//namespace oneapi
+//{
+//namespace dpl
+//{
+//namespace __par_backend_hetero
+//{
+
+template <int __block_size, int __radix, bool __is_asc, typename _Group, typename _Src, typename _Dst, typename _BinPos>
+void
+__sort_iter(_Group __wg, _Src& __src, _Dst& __dst, ::std::size_t __data_size, _BinPos& __bin_position,
+            int __radix_offset)
+{
+    constexpr int __bin_count = 1 << __radix;
+    int __histogram[__bin_count] = {0};
+    int __wi_block_pos[__bin_count];
+    int __element_pos[__block_size];
+
+    auto __wi = __wg.get_local_linear_id();
+
+    for (auto __i = 0; __i < __block_size; ++__i)
+    {
+        const auto __idx = __wi * __block_size + __i;
+        if (__idx < __data_size)
+        {
+            const int __bin = __get_bucket<(1 << __radix) - 1, __is_asc>(__to_ordered(__src[__idx]), __radix_offset);
+            // trivial exclusive scan within the WI block: for each element store the previous counter value
+            __element_pos[__i] = __histogram[__bin];
+            // and then increment the counter
+            ++__histogram[__bin];
+        }
+    }
+
+    int __bin_pos = 0; // only matters for the last WI in the group
+    for (auto __i = 0; __i < __bin_count; ++__i)
+    {
+        __wi_block_pos[__i] = sycl::exclusive_scan_over_group(__wg, __histogram[__i], sycl::plus<>());
+        if (__wi == __wg.get_local_linear_range() - 1)
+        {
+            // the last WI can iteratively do exclusive scan for bins
+            __bin_position[__i] = __bin_pos;
+            __bin_pos += __wi_block_pos[__i] + __histogram[__i]; // the reduction over all private histograms
+        }
+    }
+    sycl::group_barrier(__wg);
+
+    for (auto __i = 0; __i < __block_size; ++__i)
+    {
+        const auto __idx = __wi * __block_size + __i;
+        if (__idx < __data_size)
+        {
+            const auto __val = __src[__idx];
+            const int __bin = __get_bucket<(1 << __radix) - 1, __is_asc>(__to_ordered(__val), __radix_offset);
+            const int __new_idx = __bin_position[__bin] + __wi_block_pos[__bin] + __element_pos[__i];
+            __dst[__new_idx] = __val;
+        }
+    }
+    sycl::group_barrier(__wg);
+}
+
+template<typename _KernelName, int __block_size = 16/*from 8 to 64*/, ::std::uint32_t __radix = 4, bool __is_asc = true,
+         typename _RangeIn, typename _RangeOut>
```

Comment:
Comment says from 8 to 64, but we use from 1 to 32.   
Does the actual viable range of this depend on the device?  If so, what needs to be checked, local storage?

Reply:
Due to the code was changed,  that comment is not actual any more.